bigchaindb/elections/election.py

```diff
@@ -0,0 +1,256 @@
+            if getter(txn, 'operation') == Vote.OPERATION:
+                for output in getter(txn, 'outputs'):
+                    # NOTE: We enforce that a valid vote to election id will have only
+                    # election_pk in the output public keys, including any other public key
+                    # along with election_pk will lead to vote being not considered valid.
+                    if len(getter(output, 'public_keys')) == 1 and [election_pk] == getter(output, 'public_keys'):
+                        votes = votes + int(getter(output, 'amount'))
+        return votes
+
+    def get_commited_votes(self, bigchain, election_pk=None):
+        if election_pk is None:
+            election_pk = self.to_public_key(self.id)
+        txns = list(backend.query.get_asset_tokens_for_public_key(bigchain.connection,
+                                                                  self.id,
+                                                                  election_pk))
+        return self.count_votes(election_pk, txns, dict.get)
+
+    @classmethod
+    def has_concluded(cls, bigchain, election_id, current_votes=[], height=None):
+        """Check if the given `election_id` can be concluded or not
+        NOTE:
+        * Election is concluded iff the current validator set is exactly equal
+          to the validator set encoded in election outputs
+        * Election can concluded only if the current votes achieves a supermajority
+        """
+        election = bigchain.get_transaction(election_id)
+
+        if election:
+            election_pk = election.to_public_key(election.id)
+            votes_committed = election.get_commited_votes(bigchain, election_pk)
+            votes_current = election.count_votes(election_pk, current_votes)
+            current_validators = election.get_validators(bigchain, height)
+
+            if election.is_same_topology(current_validators, election.outputs):
+                total_votes = sum(current_validators.values())
+                if (votes_committed < (2/3)*total_votes) and \
+                        (votes_committed + votes_current >= (2/3)*total_votes):
+                    return election
+        return False
+
+    def get_status(self, bigchain):
+        concluded = self.get_election(self.id, bigchain)
+        if concluded:
+            return self.CONCLUDED
+
+        latest_change = self.get_validator_change(bigchain)
+        latest_change_height = latest_change['height']
+        election_height = bigchain.get_block_containing_tx(self.id)[0]
+
+        if latest_change_height >= election_height:
+            return self.INCONCLUSIVE
+        else:
+            return self.ONGOING
+
+    def get_election(self, election_id, bigchain):
+        result = bigchain.get_election(election_id)
+        return result
+
+    @classmethod
+    def store_election_results(cls, bigchain, election, height):
+        bigchain.store_election_results(height, election)
```

Comment:
Is nothing being returned here?

Reply:
I'll have it return a status response so I can do some error handling in the surrounding function.

Reply:
No need for a return value here. The return value for the surrounding function is handled by `on_approval`, which is specific to the type of election being called.